var/spack/repos/builtin/packages/py-breathe/package.py

```diff
@@ -42,6 +43,7 @@ class PyBreathe(PythonPackage):
     depends_on("py-sphinx@:3", type=("build", "run"), when="@:4.20")
     depends_on("py-sphinx@3:3.2", type=("build", "run"), when="@4.21:4.32")
     depends_on("py-sphinx@3:4", type=("build", "run"), when="@4.33")
-    depends_on("py-sphinx@4,5.0.1:5", type=("build", "run"), when="@4.34:")
+    depends_on("py-sphinx@4,5.0.1:5", type=("build", "run"), when="@4.34")
+    depends_on("py-sphinx@4,5.0.1:6", type=("build", "run"), when="@4.35:")
```

Comment:
```suggestion
    depends_on("py-sphinx@4,5.0.1:", type=("build", "run"), when="@4.35:")
```

Reply:
The Breathe developers specifically exclude future major versions of Sphinx from compatibility ("<7"). Historically speaking this is justified because there were changes in the Sphinx API from 4 to 5 and from 5 to 6 that broke Breathe (and other packages like `py-sphinx-design`). This restriction prevents a build error when `py-sphinx` gets a new major version and somebody tries to build an existing version of `py-breathe`, and I would respectfully suggest a stet, here.

Reply:
Historically this was hard-coded into the setup.py install_requires, but it seems like they removed the upper bound in the most recent version. I haven't followed development of this package very closely, so I'm not sure what the developers would recommend. We can add a fake upper bound if you think it's safer.